internal/ast/ast.go

```diff
@@ -10944,6 +10946,61 @@ func (node *SourceFile) ECMALineMap() []core.TextPos {
 	return lineMap
 }
 
+// GetNameTable returns a map of all names in the file to their positions.
+// If the name appears more than once, the value is -1.
+func (file *SourceFile) GetNameTable() map[string]int {
+	file.nameTableOnce.Do(func() {
+		nameTable := make(map[string]int, file.IdentifierCount)
+
+		isTagName := func(node *Node) bool {
```

Comment:
We have that function in `ls/utilities.go`, not super important but we could move it and reuse it here.